addons/account/models/account_partial_reconcile.py

```diff
@@ -119,7 +119,7 @@ def unlink(self):
         # Reverse CABA entries.
         today = fields.Date.context_today(self)
         default_values_list = [{
-            'date': move.date if move.date > (move.company_id.period_lock_date or date.min) else today,
+            'date': move._get_accounting_date(move.date, True),
```

Comment:
I don't understand all this bunch of codes... This single line is fixing Bug 1.
Bug 2 is only a traceback..
Imo your diff should be 4-5 lines, not more

Reply:
Actually my fix is 3 lines, you can see it in my commit, the other part is the back port of VIN's patch for displaying lock dates to the user. The reason that I need other changes besides this is that there were other cases provided by TSB, I have added them in tests.